src/graph/graph.c

```diff
@@ -269,26 +202,21 @@ Graph *Graph_Get(RedisModuleCtx *ctx, RedisModuleString *graph_name) {
 
 size_t Graph_NodeCount(const Graph *g) {
     assert(g);
-    return g->node_count;
+    return g->nodes->itemCount;
 }
 
-void Graph_CreateNodes(Graph* g, size_t n, int* labels, NodeIterator **it) {
+void Graph_CreateNodes(Graph* g, size_t n, int* labels, DataBlockIterator **it) {
     assert(g);
-    
-    _Graph_ResizeNodes(g, n);
-
-    if(it != NULL) {
-        *it = NodeIterator_New(GRAPH_ACTIVE_BLOCK(g),
-                               g->node_count,
-                               g->node_count + n,
-                               1);
-    }
 
-    int node_id = g->node_count;
-    g->node_count += n;
+    NodeID node_id = (NodeID)Graph_NodeCount(g);
 
+    _Graph_ResizeNodes(g, n);
     _Graph_ResizeMatrix(g, g->adjacency_matrix);
 
+    if(it) {
+        *it = DataBlockIterator_New(DataBlock_GetItemBlock(g->nodes, node_id), node_id, node_id + n, 1);
+    }
+
     if(labels) {
         for(int idx = 0; idx < n; idx++) {
```

Comment:
`idx` should probably be a `size_t` like `n`. If we're inserting > INT_MAX nodes (admittedly, eek) this condition cannot be satisfied.